csp/sync.csp

```diff
@@ -0,0 +1,187 @@
+            set filesToSync = ($length(commitMsg) > 0)
+            set fileSectionDisplay = $select(
+                filesToSync = 1: "block",
+                1: "none"
+            )
+            set noFileDisplay = $select(
+                filesToSync = 0: "block",
+                1: "none"
+            )
+
+            set outputDisplay = "none"
+
+        </server>
+        
+        <div class="container">
+            <h1 class="text-center">Sync Repository</h1>
+            <div class="row">
+                <div class="offset-sm-2 col-sm-8">
+                    <div style="display: #(fileSectionDisplay)#">
+                        <h3 class="section-header">Files to be committed with sync:</h3>
+                        <ul class="list-group">
+                            <server>
+                                set iterator = uncommittedWithAction.%GetIterator()
+                                while iterator.%GetNext(,.uncommitted) {
+                                    set action = uncommitted.%Get("action")
+                                    set file = uncommitted.%Get("file")
+                                    &html<<li class="list-group-item #(action)#">#(file)#</li>>
+                                }
+                            </server>
+                        </ul>
+                        <h3 class="section-header">Sync commit message:</h3>
+                        <input class="form-control" type="text" name="syncMsg" id="syncMsg" value="#(commitMsg)#">
+                    </div>
+                    <div style="display: #(noFileDisplay)#">
+                        <h2 class="text-center">No files to commit with sync</h2>
+                    </div>
+                    <h3 class="section-header">Sync details:</h3>
+                    <p>Upon syncing, the local repository will pull the changes from remote and commit the newest changes before committing and pushing</p>
+                    <server>
+                        
+                        if settings.defaultMergeBranch '= "" {
+                            &html<<p>Local changes will be merged with #(settings.defaultMergeBranch)# after receiving the latest changes</p>>
+                        }
+                    </server>
+
+                    <button class="btn btn-lg btn-primary" id="syncBtn" onClick="disableInput()">Sync</button>
+                    <div>
+                        <h3 class="output-header" id="outputHeader" style="display: #(outputDisplay)#">Sync output: </h3>
+                        <div class="container output" id="outputContainer" style="display: #(outputDisplay)#">
+                            <pre id="outputBox"></pre>
+                        </div>
+                    </div>
+                </div>
+            </div>
+            
+        </div>
+    </body>
+    <script type="text/javascript">
+    function disableInput() {
```

Comment:
Let's rename this function if we're going to have it actually perform the sync action

Reply:
Still pending